tools/pipelines/test-real-service.yml

```diff
@@ -198,3 +198,27 @@ stages:
                 BUILD_ID: $(Build.BuildId)
                 ADO_API_TOKEN: $(System.AccessToken)
                 PIPELINE: 'EndToEndTests'
+
+            # Retrieve E2E test pass rate
+            - task: Bash@3
+              displayName: Retrieve test pass rate
+              inputs:
+                targetType: 'inline'
+                workingDirectory: $(toolAbsolutePath)
+                script: |
```

Comment:
Let's move all this into a `.sh` file under `scripts/` at the root of the repo and update the ADO task to point to that file instead of defining it inline. And all the variables it's currently getting from ADO (the token, the target folder for the json files, the path to the handler for telemetry-generator) it should take from "generic" environment variables (e.g. `$ACCESS_TOKEN`), and the ADO task should pass the values that it has into those variables with an `env:` section.

Note as well that currently the ADO task is not passing any values in `env:` like the task above it does, and the handler code expects some of those values (`BUILD_ID`, `PIPELINE`) because it uses them when creating the telemetry event.

Reply:
Oh, can I just pass in the scripts folder to call the bash script? Something like ${{ variables.testWorkspace }}/scripts?